platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gui/GuiPlatformAccessor.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.gui;
+
+import com.swirlds.platform.Consensus;
+import com.swirlds.platform.components.state.StateManagementComponent;
+import com.swirlds.platform.gossip.shadowgraph.ShadowGraph;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Provides a way to access private platform objects from the GUI. Suboptimal, but necessary to preserve the current UI
+ * architecture if we don't want to allow public access to these objects.
+ *
+ * @deprecated this class will eventually be removed
+ */
+@Deprecated(forRemoval = true)
+public final class GuiPlatformAccessor {
+
+    private final Map<Long, String> aboutStrings = new ConcurrentHashMap<>();
+    private final Map<Long, String> platformNames = new ConcurrentHashMap<>();
+    private final Map<Long, byte[]> swirldIds = new ConcurrentHashMap<>();
+    private final Map<Long, Integer> instanceNumbers = new ConcurrentHashMap<>();
+    private final Map<Long, ShadowGraph> shadowGraphs = new ConcurrentHashMap<>();
+    private final Map<Long, StateManagementComponent> stateManagementComponents = new ConcurrentHashMap<>();
+    private final Map<Long, AtomicReference<Consensus>> consensusReferences = new ConcurrentHashMap<>();
+
+    private static final GuiPlatformAccessor INSTANCE = new GuiPlatformAccessor();
+
+    /**
+     * Get the static instance of the GuiPlatformAccessor.
+     *
+     * @return the static instance of the GuiPlatformAccessor
+     */
+    public static GuiPlatformAccessor getInstance() {
+        return INSTANCE;
+    }
+
+    private GuiPlatformAccessor() {}
+
+    /**
+     * The SwirldMain calls this to set the string that is shown when the user chooses "About" from the Swirlds menu in
+     * the upper-right corner of the window. It is recommended that this be a short string that includes the name of the
+     * app, the version number, and the year.
+     *
+     * @param nodeId the ID of the node
+     * @param about  wha should show in the "about" window from the menu
+     */
+    public void setAbout(final long nodeId, final String about) {
```

Comment:
Nullability of the String and check if nonNull.

Reply:
added